File: Admin/src/environments/environment.ts
// This file can be replaced during build by using the `fileReplacements` array.
// `ng build` replaces `environment.ts` with `environment.prod.ts`.
// The list of file replacements can be found in `angular.json`.

export const environment = {
  production: false,
  appName: 'Khanyi Solutions Admin',
  appVersion: '1.0.0',

  // Khanyi API Configuration
  apiUrl: 'http://localhost:3000/api/v1',
  apiTimeout: 30000,

  // Authentication
  defaultauth: 'khanyi', // Changed from 'fackbackend' to our auth
  tokenKey: 'khanyi_admin_token',
  userKey: 'khanyi_admin_user',

  // Features
  features: {
    enableAnalytics: true,
    enableNotifications: true,
    enableReports: true,
    enableBulkOperations: true,
    enableRealTimeUpdates: true
  },

  // System limits
  limits: {
    maxFileSize: 10485760, // 10MB
    maxBulkOperations: 1000,
    rateLimitPerMinute: 100
  },

  // Firebase Config (keep for existing template features if needed)
  firebaseConfig: {
    apiKey: 'AIzaSyAIrRUP8z39yFhjsfnLsxwDhzxzAguzHvI',
    authDomain: 'vixon-angular.firebaseapp.com',
    databaseURL: '',
    projectId: 'vixon-angular',
    storageBucket: 'vixon-angular.appspot.com',
    messagingSenderId: '822277458131',
    appId: '1:822277458131:web:5ec3bee2543b96567aac26',
    measurementId: 'G-YPD27EXC6H'
  }
};



/*
 * For easier debugging in development mode, you can import the following file
 * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
 *
 * This import should be commented out in production mode because it will have a negative impact
 * on performance if an error is thrown.
 */
// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.

File: lib/config/environment.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';

class Environment {
  static String get apiBaseUrl =>
      dotenv.env['API_BASE_URL'] ?? 'http://192.168.1.104:3000';
  static String get appName => dotenv.env['APP_NAME'] ?? 'Khanyi Vending App';
  static String get appVersion => dotenv.env['APP_VERSION'] ?? '1.0.0';

  // Features
  static bool get enableAnalytics =>
      dotenv.env['ENABLE_ANALYTICS']?.toLowerCase() == 'true';
  static bool get enableNotifications =>
      dotenv.env['ENABLE_NOTIFICATIONS']?.toLowerCase() == 'true';
  static bool get enableReports =>
      dotenv.env['ENABLE_REPORTS']?.toLowerCase() == 'true';

  // System Limits
  static int get maxFileSize =>
      int.tryParse(dotenv.env['MAX_FILE_SIZE'] ?? '10485760') ?? 10485760;
  static int get maxBulkOperations =>
      int.tryParse(dotenv.env['MAX_BULK_OPERATIONS'] ?? '1000') ?? 1000;
  static int get rateLimitPerMinute =>
      int.tryParse(dotenv.env['RATE_LIMIT_PER_MINUTE'] ?? '100') ?? 100;
}
